packages/ai-llamafile/src/node/llamafile-manager-impl.ts

```diff
@@ -0,0 +1,92 @@
+// *****************************************************************************
+// Copyright (C) 2024 EclipseSource GmbH.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+import { LanguageModelRegistry } from '@theia/ai-core';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { ChildProcessWithoutNullStreams, spawn } from 'child_process';
+import { basename, dirname } from 'path';
+import { fileURLToPath } from 'url';
+import { LlamafileLanguageModel } from '../common/llamafile-language-model';
+import { LlamafileEntry, LlamafileManager, LlamafileServerManagerClient } from '../common/llamafile-manager';
+
+@injectable()
+export class LlamafileManagerImpl implements LlamafileManager {
+
+    @inject(LanguageModelRegistry)
+    protected languageModelRegistry: LanguageModelRegistry;
+
+    private processMap: Map<string, ChildProcessWithoutNullStreams> = new Map();
+    private client: LlamafileServerManagerClient;
+
+    addLanguageModels(llamaFiles: LlamafileEntry[]): void {
+        const models = llamaFiles.map(llamafile =>
+            new LlamafileLanguageModel(llamafile.name, llamafile.uri, llamafile.port));
+        this.languageModelRegistry.addLanguageModels(models);
```

Comment:
This unconditionally adds language models to the registry. However this method can be called mutliple times, e.g. when there are mutliple frontends open. We should not add duplicates to the registry.